sim/midas/src/main/scala/midas/passes/xilinx/package.scala

```diff
@@ -15,7 +15,7 @@ package object xilinx {
 
   object ReplaceAbstractClockGates extends Transform with NoAnalysisPass {
     val transformer = StatementTransformer {
-      case wi: WDefInstance if wi.module == DefineAbstractClockGate.blackboxName =>
+      case wi: WDefInstance if wi.module.contains(DefineAbstractClockGate.blackboxName) =>
```

Comment:
Would it make sense to check the extmodule's `defname` instead? That can be picked pretty freely, and in case multiple extmodule declarations exist with deduplicated names, the `defname` would still be identical among all of them.

I guess the downside is that you'd have to go lookup the extmodule definition here instead of being able to tell from the `WDefInstance` that the module is a clock gate :thinking:

Reply:
Cool!